src/osx/core/uilocale.mm

```diff
@@ -172,7 +174,7 @@ int CompareStrings(const wxString& lhs, const wxString& rhs,
                     str = [m_nsloc localizedStringForLanguageCode:[m_nsloc languageCode]];
                     break;
                 case wxLOCALE_FORM_ENGLISH:
-                    str = [m_nsloc displayNameForKey:NSLocaleIdentifier value:[m_nsloc localeIdentifier]];
+                    str = [enLocale displayNameForKey:NSLocaleIdentifier value:[m_nsloc localeIdentifier]];
```

Comment:
According to @utelle's changes in #23199 this should be
```suggestion
                    str = [enLocale displayNameForKey:NSLocaleLanguageCode value:[m_nsloc localeIdentifier]];
```

Reply:
Although it works when using `localeIdentifier` for the value attribute it would be clearer to use `languageCode`:
```c++
str = [enLocale displayNameForKey:NSLocaleIdentifier value:[m_nsloc languageCode]];
```
Even more concise would be:
```c++
str = [enLocale localizedStringForLanguageCode:[m_nsloc languageCode]];
```
Therefore I would recommend to use the latter.